D[2] CLI docs: Cascade options with CLI (#2760)
webpack/webpack.js.org: src/content/api/cli.md

| `--output-library-target`      | The type for exposing the exports of the entry point as library | string     | var                                                                             |
| `--output-path`                | The output path for compilation assets                          | string     | Current directory                                                               |
| `--output-pathinfo`            | Include a comment with the request for every dependency         | boolean    | false                                                                           |
| `--output-public-path`         | The public path for the assets                                  | string     | /                                                                               |
| `--output-source-map-filename` | The output filename for the SourceMap                           | string     | [name].map or [outputFilename].map                                              |
| `--build-delimiter`            | Display custom text after build output                          | string     | Default string is null. You could provide a string such as `=== Build done ===` |

### Cascade Options

> Sometime we want to add a option conditionally to an existing script, without writing writing the same script again so we can cascade option to it with just using `--` before the option and access it in our config file.

Reviewer: ```suggestion
CLI allows to provide variables by adding `--` before the variable name and value. You can access such variables in the configuration file.
```

For npm:

```bash
"build": "webpack --mode production --config webpack.prod.config.js"
"build:compress": "npm run build -- --env.presets compress",
```

For yarn, we can skip `--` :

```bash
"build:compress": "yarn build --env.presets compress",
```

> 

Reviewer: it lacks javascript example where you read the variable in config

Author: Great Suggestion. I will do it.

Reviewer: You want to do it in this iteration right?

| `--output-path`                | The output path for compilation assets                          | string     | Current directory                                                               |
| `--output-pathinfo`            | Include a comment with the request for every dependency         | boolean    | false                                                                           |
| `--output-public-path`         | The public path for the assets                                  | string     | /                                                                               |
| `--output-source-map-filename` | The output filename for the SourceMap                           | string     | [name].map or [outputFilename].map                                              |
| `--build-delimiter`            | Display custom text after build output                          | string     | Default string is null. You could provide a string such as `=== Build done ===` |

### Cascade Options

CLI allows to provide variables by adding `--` before the variable name and value. You can access such variables in the configuration file.

For npm:

```bash

> "build": "webpack --mode production --config webpack.prod.config.js"

Reviewer: these are not bash commands, looks like scripts from package.json declaration, maybe you want to correct it to either be bash command or change it to package.json example? Although we dont really do package.json examples outside of guides so i'd vote for 1st

"build:compress": "npm run build -- --env.presets compress",
```

For yarn, we can skip `--` :

```bash

> "build:compress": "yarn build --env.presets compress",

Reviewer: This wont run, maybe you meant:
```suggestion
yarn build --env.presets compress
```

```

> T> Hence you can get only `{ presets: "compress" }` in [environment variables](/guides/environment-variables) when you run `build:compress`.

Reviewer: We use single quotes when demonstrate JavaScript
```suggestion
T> Hence you can get only `{ presets: 'compress' }` in [environment variables](/guides/environment-variables/) when you run `build:compress`.
```

> In webpack config you can access the variable as follows:

Reviewer: ```suggestion
You can access the variable in webpack configuration by reading it from the `env` object which gets passed down to it by webpack:
```
